libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -2044,6 +2048,62 @@ internal class MapboxNavigationTest : MapboxNavigationBaseTest() {
         }
     }
 
+    @Test
+    fun setRerouteEnabled() {
+        val customRerouteController = mockk<NavigationRerouteController>(relaxed = true)
+        val defaultRerouteController = mockk<InternalRerouteController>(relaxed = true)
+        every {
+            NavigationComponentProvider.createRerouteController(
+                any(),
+                any(),
+                any(),
+                any(),
+                any(),
+                any()
+            )
+        } returns defaultRerouteController
+        every { defaultRerouteController.state } returns RerouteState.FetchingRoute
+        every { customRerouteController.state } returns RerouteState.FetchingRoute
+        createMapboxNavigation()
+
+        assertNotNull(mapboxNavigation.getRerouteController())
+        assertTrue(mapboxNavigation.isRerouteEnabled())
+
+        mapboxNavigation.setRerouteEnabled(false)
+
+        assertNull(mapboxNavigation.getRerouteController())
+        assertFalse(mapboxNavigation.isRerouteEnabled())
+        verify { defaultRerouteController.interrupt() }
+        clearAllMocks(answers = false)
+
+        mapboxNavigation.setRerouteEnabled(true)
+
+        assertEquals(defaultRerouteController, mapboxNavigation.getRerouteController())
+        assertTrue(mapboxNavigation.isRerouteEnabled())
+
+        mapboxNavigation.setRerouteController(customRerouteController)
+        verify {
+            defaultRerouteController.interrupt()
+            customRerouteController.reroute(any<NavigationRerouteController.RoutesCallback>())
+        }
+        clearAllMocks(answers = false)
+        val customRerouteControllerAdapter = mapboxNavigation.getRerouteController()
+        assertNotEquals(defaultRerouteController, customRerouteControllerAdapter)
```

Comment:
That's interesting:
```
mapboxNavigation.setRerouteController(customRerouteController)
assertEquals(customRerouteController, mapboxNavigation.getRerouteController()) // fails
```
I don't think it's needed to be fixed. Just wanted to share the observation 🙂

Reply:
That's right, because I added an adapter.